src/lib/support/BufferWriter.h

```diff
@@ -136,6 +143,7 @@ class BufferWriter : public BufferWriterBase<BufferWriter>
         }
         return *this;
     }
+    void Reset(uint8_t * buf, size_t len) { BufferWriterBase<BufferWriter>::Reset(buf, len); }
```

Comment:
Why not just make `Reset` public on the superclass instead?

Reply:
If we require it public, then we may as well do so.

I did not provide a Reset public on purpuse when I originally developed it, however that may have been because I used to have a 'From' method that was converting little to big endian writers while preserving 'mNeeded'. That was removed because it caused headaches though ... so we could make it public.

We have operator = though, so instead of:

```cpp
writer.Reset(buffer, len);
```

i could do:

```cpp
writer = LittleEndian::BufferWriter(buffer, len);
```

without code changes.

Reply:
Where are we currently using reset? I always assumed there is very little need for resetting buffers.

Reply:
`PacketBufferWriter` is a subclass and needs to set and clear the `BufferWriter` fields. But yes, `=` works, so `Reset` can go.